packages/client/hmi-client/src/services/model.ts

```diff
@@ -116,9 +118,36 @@ export function validateModelName(name: string): boolean {
 	return true;
 }
 
-export async function profile(modelId: string, documentId: string): Promise<Model | null> {
+export async function profile(modelId: string, documentId: string): Promise<string | null> {
 	const profileModelJobId = await profileModel(modelId, documentId);
 	await fetchExtraction(profileModelJobId);
-	const model = await getModel(modelId);
-	return model;
+	return modelId;
+}
+
+/**
+ * Generates a model card based on the provided document ID, model ID, and model service type.
+ *
+ * @param {string} documentId - The ID of the document.
+ * @param {string} modelId - The ID of the model.
+ * @param {ModelServiceType} modelServiceType - The type of the model service.
+ *
+ * @returns {Promise<string | null>} The ID of the model if the operation is successful, null otherwise.
```

Comment:
This function should not return anything.
We should just display that they were an error that happened via popup.

Reply:
Changed